pkg/mesos/taskio.go

```diff
@@ -298,30 +297,19 @@ func (t *TaskIO) attachContainerInput() error {
 			for {
 				// Use a buffer with a reasonable size in order to perform well when STDIN is piped and has
 				// a significant size. See https://jira.mesosphere.com/browse/DCOS_OSS-5357
-				bufLen := 512
-
-				if t.opts.TTY {
-					// TODO(bamarni): investigate on why "test_task:test_attach" fails if the buffer size below
-					// is greater than 1, this looks like an issue with the term package we're using, which doesn't
-					// work properly when reading more than 1 char at a time.
-					bufLen = 1
-				}
-				buf := make([]byte, bufLen)
+				buf := make([]byte, 512)
 				n, err := t.opts.Stdin.Read(buf)
-				if _, ok := err.(term.EscapeError); ok {
-					t.cancelFunc()
-					t.exitSequenceDetected = true
-					return
```

Comment:
This change is unrelated, there were a race condition between sending an empty input data to indicate EOF and calling cancelFunc(), which can cause the input connection to be called before Mesos received the EOF message.

Moving cancelFunc() to after the input HTTP call finished (once the acif channel is closed) makes sure Mesos has received EOF on STDIN and we only cancel the call for container output.